hbase-common/src/main/java/org/apache/hadoop/hbase/ServerName.java

```diff
@@ -143,11 +143,10 @@ public static ServerName valueOf(final String hostname, final int port, final lo
    * instances, though we may return a shared immutable object as an internal optimization.
    */
   public static ServerName valueOf(final String serverName) {
-    int firstSep = serverName.indexOf(SERVERNAME_SEPARATOR);
-    int lastSep = serverName.lastIndexOf(SERVERNAME_SEPARATOR);
-    String hostname = serverName.substring(0, firstSep);
-    int port = Integer.parseInt(serverName.substring(firstSep + 1, lastSep));
-    long startCode = Long.parseLong(serverName.substring(lastSep + 1));
+    String[] splits = serverName.split(",");
```

Comment:
ServerName is IA.Public so we should not change the behavior of its method.

So how do other shell commands deal with this problem?

Reply:
The cause of the problem started from this issue https://issues.apache.org/jira/browse/HBASE-24305，The ServerName function previously supported filling in only the host name and port。Currently, it also affects clear_ Compaction_ Queues.rb。Other RBs do not support the forms of host name and port in promt。

Reply:
Should we maintain their consistency？
1. If the serverName supports the form of host name and port, then the relevant client methods can also maintain consistency.
2. Remove support from compact_ rs and clear_ compaction_ queues，It doesn't have much effect either.

Of course, maybe we have other plans. Do you have any ideas?

Reply:
Checked the code change in HBASE-24305, seems the problem is that, before HBASE-24305, if we pass a String with only host and port, it will use port as start code to create the server name, which is obviously incorret but it could work if you just do not care about the start code...

So I suggest that, we introduce a new method, explicitly say that we can pass ServerName without start code, and then you can use it if you do not care about start code.

WDYT?

Reply:
I would suggest modifying the compact_ rs annotation which can supports hostname:port or hostname,port,startCode with more reasonable semantics. Internally, we can support it by using the existing parseServerName.